### ISSUE TITLE
lua 5.2 - remove getfenv in base.lua

### ISSUE THREAD

**Issue by davidm** [fab13n/metalua#12]
For Lua 5.2 compatibility, remove getfenv in base.lua.

**Comment by agladysh** [comment]
Whatever you do, please do keep MetaLua compatible with 5.1

**Comment by fab13n** [comment]
I plan to overhaul this in the repackaging branch, together with cutting
metalua into independent libs.

On Wed, Jan 25, 2012 at 8:41 AM, David Manura <
reply@reply.github.com

> wrote:
> 
> For Lua 5.2 compatibility, remove getfenv in base.lua.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/fab13n/metalua/issues/12

## 

Fabien Fleutot
+---
| 33 chemin de Mange-Pommes
| 31520 Ramonville Saint-Agne -- France
| mobile: +33 6 28 06 09 97
| office: +33 5 61 00 06 49
| home: +33 5 61 75 05 67

**Comment by davidm** [comment]
Yes, IMHO, Metalua should run on both 5.1 and 5.2 and parse/compile both 5.1 and 5.2 (including 5.2 from 5.1 and vice-versa).

BTW, the lexer currently lexes a superset of 5.1 and 5.2, but I'm not aware of 5.2 support in the parser yet.  We'll need to think about how best to handle the 5.2 migration.  See also http://lua-users.org/wiki/LuaVersionCompatibility for ideas.

**Comment by alexbw** [comment]
+1
